### PR TITLE
Extract hashes of translation strings into included modules

### PR DESCRIPTION
**Pull request by lizconlan**
Otherwise the default locale translations are always returned instead of the translations for the requested locale. 

Example:

```rb
I18n.with_locale(:fr) { _("Freedom of Information Act") }
=> "Loi pour la liberté d'information"

I18n.with_locale(:fr) { InfoRequest.first.law_used_human(:act) }
=> "Freedom of Information Act"
```

Connects to mysociety/belgium-theme#45 (will need a hotfix for version 0.28, and any theme that overrides `LAW_USED_READABLE_DATA` will need to be updated - may only affect AskTheEU, currently on 0.25 but worth noting in the Upgrade Notes)

### CODE REVIEW

File: app/models/info_request.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ class InfoRequest < ActiveRecord::Base
   include Rails.application.routes.url_helpers
 
   # Two sorts of laws for requests, FOI or EIR
-  LAW_USED_READABLE_DATA =
+  def law_used_readable_data
```
Comment (lizconlan) [inline app/models/info_request.rb]: should this be a private method? I think we can still override it in themes if it is

Comment (garethrees) [inline app/models/info_request.rb]: Yeah, +1 for private.

File: spec/models/info_request_spec.rb [hunk 2]
```diff
@@ -1209,6 +1209,15 @@
           with_message( "Unknown key 'random' for '#{info_request.law_used}'")
       end
 
+      context 'using the non default language' do
```
Comment (garethrees) [inline spec/models/info_request_spec.rb]: I actually don't think we should test for this.

I'd _expect_ this to just work. I wouldn't write a spec each time we added a translated string, so I don't think we should here.

A better thing to do would be to add a custom hound check to assert that we haven't assigned translated values to constants.

**Comment by garethrees** [comment]
> will need a hotfix for versions 0.26 to 0.28

We only really [bugfix against the current version](http://alaveteli.org/docs/running/security/#bug-fixes). It would be better to just upgrade sites that _really_ want this fix.

**Comment by garethrees** [comment]
> I think we should make one commit that fixes this problem across the whole app.

Here's a [quick suggestion](https://gist.github.com/anonymous/1ad5dc5bf28b87d143e5538972bddea7). I'm not set on this, but I can't think of an elegant way to define these before the macro methods that doesn't look like there's a random class method [dumped](https://github.com/bbatsov/rails-style-guide#macro-style-methods) at the top of the file.

File: spec/models/info_request_spec.rb (hunk 2 above)
Comment (lizconlan) [inline spec/models/info_request_spec.rb]: good point, although AFAIK this uses the translation file in spec/fixtures/locale so it shouldn't be affected by Transifex changes (shouldn't)

**Comment by lizconlan** [comment]
>> will need a hotfix for versions 0.26 to 0.28
>
> We only really bugfix against the current version. It would be better to just upgrade sites that really want this fix.

Currently causing a problem for Belgium (0.27) so maybe a 0.27 hotfix to avoid retranslating so soon. Will have a proper look next week

File: spec/models/info_request_spec.rb (hunk 2 above)
Comment (lizconlan) [inline spec/models/info_request_spec.rb]: confirmed - the spec uses a fixture file for translations

**Comment by lizconlan** [comment]
PR against develop and a hotfix for 0.27 to avoid making Belgium redo their translations?

**Comment by lizconlan** [comment]
@garethrees does fdde543 look ok?

**Comment by lizconlan** [comment]
hotfix for 0.28

**Comment by lizconlan** [comment]
Happy to squash some or all of those post review (although I'd be inclined to keep the Pro one separate)

File: app/models/mail_server_log/translated_constants.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# -*- encoding : utf-8 -*-
+class MailServerLog::DeliveryStatus
```
Comment (garethrees) [inline app/models/mail_server_log/translated_constants.rb]: The path to this file should be `app/models/mail_server_log/delivery_status/translated_constants.rb`

**Comment by lizconlan** [comment]
ToDo - cherry pick 23a7e4f onto a new branch off master to create a hotfix comment (holding off in expectation of a translation hotfix which gets priority)

**Comment by lizconlan** [comment]
Closed in favour of a hotfix branch #3929

**Pull request closed without merging**